### PR TITLE
client: Make gRPC INTERNAL status retryable

### PR DESCRIPTION
**Pull request by mjameswh**
## What was changed and why?

- In client, some variants of gRPC's `INTERNAL` status code are now considered as retryable (more specifically: errors where message contain `RST_STREAM with code 0` or `RST_STREAM with code 2`, except TLS related issues). Users are seeing these specific errors quite regularly, apparently due to transient issues at the HTTP2 connection level. After discussion with both SDK and Server team, we believe that in most case retrying these errors would be beneficial, and in the worst case, should only result in minor waste of resource on the server.

- In WorkflowClient, `list().withHistories()` no longer swallow client errors (they would previously result in an entry where `history` is `undefined`, with no way to know the underlying issue).

### CODE REVIEW

File: packages/client/src/grpc-retry.ts [hunk 1]
```diff
@@ -100,6 +100,7 @@ export function defaultGrpcRetryOptions(options: Partial<BackoffOptions> = {}):
 const retryableCodes = new Set([
   grpc.status.UNKNOWN,
   grpc.status.RESOURCE_EXHAUSTED,
+  grpc.status.INTERNAL,
   grpc.status.UNAVAILABLE,
   grpc.status.ABORTED,
```
Comment (Sushisource) [inline packages/client/src/grpc-retry.ts:105]: Wait, if it's supposed to be becoming _retryable_, and this is the list of _retryable_ codes...

**Comment by Quinn-With-Two-Ns** [comment]
>gRPC's INTERNAL status code is now considered as retryable. That's what we do in all other SDKs.

I don't think this is true. All SDKs do **not** consider INTERNAL retryable.

File: packages/client/src/grpc-retry.ts (hunk 1 above)
Comment (Sushisource) [reply packages/client/src/grpc-retry.ts:105]: I can't read. Ignore me.

**Review by Sushisource — APPROVED** [review]

**Comment by cretz** [comment]
:+1: to @Quinn-With-Two-Ns - internal is not meant to be retried IIUC (except on older servers based on capability, but that was a long time ago, can probably just not retry).

**Pull request merged**